docs/post-mortems/2019-03-26.md

```diff
@@ -0,0 +1,312 @@
+# March 26th, 2019 Boskos Outage
+
+All times in PDT (UTC-7).
+
+## Summary
+
+The Boskos instance responsible for vending GCP projects to test jobs got stuck in a crash loop
+due to a hidden bug that was unexpectedly deployed at night. Consequently, all jobs that needed
+a project from boskos (including a required job for all kubernetes/kubernetes PRs) failed for
+just over nine hours, while on-call was sleeping.
+
+## Impact
+
+All jobs depending on boskos failed between 2019-02-25 23:04 and 2019-02-26 08:12. As a result
+it was impossible to merge any PR in kubernetes/kubernetes during those nine hours, because
+`pull-kubernetes-e2e-gce` would always fail. There were similar effects on other projects during
+this time period. On `pull-kubernetes-e2e-gce` alone, 70 failures were caused by this outage.
+
+## Root Cause
+
+A [bug was introduced](https://github.com/kubernetes/test-infra/commit/02110870507abdeb93dfd6205c70f82145f8993c#r32917192)
+in a commit that landed in December last year, that when deployed would cause Boskos to be unable
+to start up. Nobody actually attempted to deploy Boskos until last week (2019-03-20). However, while
+the StatefulSet that runs Boskos was successfully updated, the actual pod was not replaced due to
+lack of any configuration to do so, and nobody noticed the lack of functional change.
+
+At around 17:20, several hours before the outage, we started upgrading the Prow clusters from Kubernetes
+1.11 to 1.13. The build cluster has 150 nodes, and each update takes five minutes, so for this entire
+process to complete would take about ten hours. Several hours later, nothing looked amiss, and we
+went to bed.
+
+At 23:04 the node running the only Boskos pod was drained to be replaced with a 1.13 node, and boskos
+was moved to a new node. At this point it used the StatefulSet's spec, which specified a newer image
+version with the bug. As a result, boskos was never able to come up on the new node.
+
+## Detection
```

Comment:
https://github.com/kubernetes/test-infra/issues/11953?

Reply:
Added a link to the AIs.